src/wallet/mod.rs

```diff
@@ -619,9 +619,25 @@ where
             _ => 1,
         };
 
+        let last_sync_height = self
+            .database()
+            .get_sync_time()?
+            .map(|sync_time| sync_time.block_time.height);
+
+        // If they didn't tell us the current height, we assume it's the latest sync height.
+        let current_height = params.current_height.or(last_sync_height);
```

Comment:
It's not the end of the world, but maybe you could move the database lookup inside an `or_else` closure, so that it happens only if it's actually required. Otherwise you may fetch the last sync height and then throw it away because there was a custom `curent_height` set.

Reply:
While this is technically possible, it's not really ideal:

- Either I completely discard whatever error get_sync_time throws to me:
```
       let current_height = params.current_height.or_else(|| {
           self.database()
               .get_sync_time()
               .ok()?
               .map(|sync_time| sync_time.block_time.height)
       });

```

- Or I turn this simple expression into a functional mess:
```
        let current_height = params.current_height.map_or_else(|| -> Result<_, Error> {
            Ok(self.database()
                .get_sync_time()?
                .map(|sync_time| sync_time.block_time.height))
        }, |x| Ok(Some(x)))?;
```